flowbased-computation/flowbased-computation-impl/src/main/java/com/farao_community/farao/flowbased_computation/impl/LoopFlowUtil.java

```diff
@@ -0,0 +1,154 @@
+import com.farao_community.farao.flowbased_computation.FlowBasedComputationResult;
+import com.farao_community.farao.flowbased_computation.glsk_provider.GlskProvider;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.iidm.network.Country;
+import com.powsybl.iidm.network.Network;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Loop flow util is used
+ * - for loop flow calculation (used in search tree rao for CCR Core), and
+ * - for Min RAM adjustment calculation
+ * @author Pengbo Wang {@literal <pengbo.wang at rte-international.com>}
+ */
+public final class LoopFlowUtil {
+    private LoopFlowUtil() {
+    }
+
+    /**
+     * calculate loop flow
+     * input CracFile
+     * This function could be removed when Flowbased computation no longer use CracFile but Crac
+     */
+    public static Map<String, Double> calculateLoopFlows(Network network,
+                                                         CracFile cracFile,
+                                                         GlskProvider glskProvider,
+                                                         List<String> countries,
+                                                         ComputationManager computationManager,
+                                                         FlowBasedComputationParameters flowBasedComputationParameters
+    ) {
+        //Call flowbased computation on given glskProvider
+        FlowBasedComputationProvider flowBasedComputationProvider = new FlowBasedComputationImpl();
+        FlowBasedComputationResult flowBasedComputationResult = flowBasedComputationProvider.run(network,
+                cracFile, glskProvider, computationManager, network.getVariantManager().getWorkingVariantId(), flowBasedComputationParameters)
+                .join();
+
+        Map<String, Double> frefResults = frefResultById(flowBasedComputationResult); //get reference flow
+        Map<String, Map<String, Double>> ptdfResults = ptdfResultById(flowBasedComputationResult); // get ptdf
+        Map<String, Double> referenceNetPositionByCountry = getRefNetPositionByCountry(network, countries); // get Net positions
+        return buildLoopFlowsFromResult(flowBasedComputationResult, frefResults, ptdfResults, referenceNetPositionByCountry);
+    }
+
+    /**
+     * calculate loop flow
+     * input Crac
+     */
+    public static Map<String, Double> calculateLoopFlows(Network network,
+                                                         Crac crac,
+                                                         GlskProvider glskProvider,
+                                                         List<String> countries,
+                                                         ComputationManager computationManager,
+                                                         FlowBasedComputationParameters flowBasedComputationParameters
+    ) {
+        //Call flowbased computation on given glskProvider
```

Comment:
`todo : use FlowBasedComputation api to find a provider, once it is based only on Crac and not on CracFile`

Reply:
ok